TechnicalDocuments/Proposals/ScreenLifecycleObservation.md

```diff
@@ -0,0 +1,67 @@
+# Generalized Screen Lifecycle Observation
+
+* Author(s): Anders Ha
+* Review Manager: Sergey Shulga
+* Implementation: [Bento#93](https://github.com/Babylonpartners/Bento/pull/93)
+
+## Introduction
+
+The proposal introduces a generalized mechanism for screen lifecycle observation.
+
+## Motivation
+
+It has been observed that there is an increasing need for the screen interaction logic to react to screen lifecycle changes.
+
+For example:
+
+* A screen might want to emit an analytics event when it is first presented.
+* A screen might want to reload its data when it appears on the screen again.
+* A screen might want to perform side effects when the screen disappears indefinitely.
+
+## Proposed solution
+
+Introduce a screen lifecycle observing mechanism which BentoKit screens may participate by conforming `BoxViewModel` to `ScreenLifecycleAware`.
+
+```swift
+extension ExampleViewModel: BoxViewModel {
+    func send(_ event: ScreenLifecycleEvent) {
+        if event.status == .appearing && event.isBeingPresentedInitially {
+            analytics.track(ExampleAnalyticsEvent.pageView)
+                   
+            // [Optional] Screen state machine needs to be aware of certain lifecycle changes.
+            actionsObserver.send(value: .screenIsAppearingInitially)
+        }
+    }
+}
+```
+
+The event model `ScreenLifecycleEvent` reflects the lifecycle status of a given screen at a given time. The status can be directly mapped to messages emitted by UIKit:
+
+Status | UIKit message
+--- | ---
+`willAppear` | `viewWillAppear`
+`didAppear` | `viewDidAppear`
+`willDisappear` | `viewWillDisappear`
+`didDisappear` | `viewDidDisappear`
```

Comment:
I think we would need to `viewDidLoad` as well. This is the best time in the view life cycle to assign the accessibility identifier for the view.